test/object-store/migrations.cpp

```diff
@@ -3032,280 +2258,278 @@ TEST_CASE("migration: AdditiveDiscovered") {
-
-            REQUIRE_NOTHROW(realm->update_schema(set_indexed(schema, "object", "value", false)));
-            REQUIRE(table->has_search_index(col_keys[0]));
-
-            REQUIRE_NOTHROW(realm->update_schema(set_indexed(schema, "object", "value 2", true)));
-            REQUIRE(!table->has_search_index(col_keys[1]));
-        }
+    TestFile config;
+    config.cache = false;
+    config.schema = schema;
+    config.schema_mode = GENERATE(SchemaMode::AdditiveDiscovered, SchemaMode::AdditiveExplicit);
+    auto realm = Realm::get_shared_realm(config);
+    REQUIRE_UPDATE_SUCCEEDS(*realm, schema, 0);
 
-        DYNAMIC_SECTION("can remove properties from existing tables, but column is not removed" << mode_string) {
-            auto table = ObjectStore::table_for_object_type(realm->read_group(), "object");
-            REQUIRE_NOTHROW(realm->update_schema(remove_property(schema, "object", "value")));
-            REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object")->get_column_count() == 2);
-            auto const& properties = realm->schema().find("object")->persisted_properties;
-            REQUIRE(properties.size() == 1);
-            auto col_keys = table->get_column_keys();
-            REQUIRE(col_keys.size() == 2);
-            REQUIRE(properties[0].column_key == col_keys[1]);
-        }
+    INFO((config.schema_mode == SchemaMode::AdditiveDiscovered ? "AdditiveDiscovered" : "AdditiveExplicit"));
 
-        DYNAMIC_SECTION("cannot change existing property types" << mode_string) {
-            REQUIRE_THROWS(realm->update_schema(set_type(schema, "object", "value", PropertyType::Float)));
-        }
+    SECTION("can add new properties to existing tables") {
+        REQUIRE_NOTHROW(realm->update_schema(add_property(schema, "object", {"value 3", PropertyType::Int})));
+        REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object")->get_column_count() == 3);
+    }
 
-        DYNAMIC_SECTION("cannot change existing property nullability" << mode_string) {
-            REQUIRE_THROWS(realm->update_schema(set_optional(schema, "object", "value", true)));
-            REQUIRE_THROWS(realm->update_schema(set_optional(schema, "object", "value 2", false)));
-        }
+    SECTION("can add new tables") {
+        REQUIRE_NOTHROW(realm->update_schema(add_table(schema, {"object 2",
+                                                                {
+                                                                    {"value", PropertyType::Int},
+                                                                }})));
+        REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object"));
+        REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object 2"));
+    }
 
-        DYNAMIC_SECTION("cannot change existing link targets" << mode_string) {
+    SECTION("embedded orphan types") {
+        if (config.schema_mode == SchemaMode::AdditiveDiscovered) {
```

Comment:
The AdditiveExplicit case here is just a schema validation error and is tested in the tests for that.